### ISSUE TITLE
mkPoetryEnv without development dependencies

### ISSUE THREAD

**Issue by yajo** [nix-community/poetry2nix#627]
### Describe the issue

While packaging an Env for further reuse, the development dependencies end up in the final package, but are not needed for production.

There should be a flag to avoid including them.

### Additional context

I don't think this is strictly required because it happens on any project that has any development dependencies. Just putting here some samples:

- `default.nix`/`shell.nix`/`flake.nix`
```
poetry2nix.mkPoetryEnv {
    src = ./.;
};
```

- `pyproject.toml`
```
# ...
[tool.poetry.dev-dependencies]
black = "^22.3.0"
```

**Comment by andersk** [comment]
Poetry 1.2.0a2 [generalizes](https://python-poetry.org/blog/announcing-poetry-1.2.0a2/#dependency-groups) the `dependencies` vs. `dev-dependencies` split to arbitrary groups (so you might have `prod`, `dev`, `test`, `docs`, for example). It’d be nice for poetry2nix to provide ways to install a specific set of groups.

**Comment by RaitoBezarius** [comment]
AFAIK, `mkPoetryEnv` has a `groups` parameter, which defaults to `[ "dev" ]`, so, is it still an issue for you @yajo @andersk ?

**Comment by yajo** [comment]
AFAIK this is fixed since poetry 1.2 support landed. I'll close, we can reopen later otherwise.

**Comment by nikhiljha** [comment]
This appears to still be broken, I have `groups = []` but it still installs development dependencies.

**Comment by yajo** [comment]
I fixed it in my case adding `checkGroups = [];` to the call to `mkPoetryApplication`. This is not documented though, so it could be one fix.

**Comment by bobvanderlinden** [comment]
I also ran into this issue. I tried the same for `mkPoetryEnv`, but there it fails: `called with unexpected argument 'checkGroups'`.

**Comment by cpcloud** [comment]
Closed by #1648
